lib/app/pages/books_screen.dart

```diff
@@ -37,22 +45,47 @@ class _BookScreenState extends ConsumerState<BookScreen> {
               children: [
                 bookState.loading
                     ? const CircularProgressIndicator()
-                    : Column(
-                        children: bookState.books
+                    : Column(children: [
+                        DropdownButton<String>(
+                          hint: Text(selectedSubject ?? 'sujet'),
+                          onChanged: (value) async {
+                            setState(() {
+                              selectedSubject = value;
+                            });
+                            await ref
```

Comment:
This may be a react-influenced viewpoint, but I would expect the filtering to happen in response to a change in the state, rather than imperatively following a selection in the dropdowns. For example, consider the case where the latest filter is persisted in local storage, and restored when the app starts.